src/types.rs

```diff
@@ -399,4 +437,73 @@ mod test {
         assert!(AstarteType::String("hello".to_owned()) == "hello");
         assert!(AstarteType::BinaryBlob(vec![1, 2, 3, 4]) == vec![1_u8, 2, 3, 4]);
     }
+
+    // we test TryFrom<AstarteType> to all the base types, using this two macro
+    macro_rules! test_reverse_type_conversion_traits_simple {
+        ($(($astartetype:tt, $typ:tt, $d:expr),)*) => {
+            $(
+                let data = $d;
+                let a_data = AstarteType::$astartetype(data.clone());
+                assert_eq!($typ::try_from(a_data)?, data);
+            )*
+        }
+    }
+    macro_rules! test_reverse_type_conversion_traits_complex {
+        ($(($astartetype:tt, $typ:tt, $subtyp:ty, $d:expr),)*) => {
+            $(
+                let data = $d;
+                let a_data = AstarteType::$astartetype(data.clone());
+                assert_eq!($typ::<$subtyp>::try_from(a_data)?, data);
+            )*
+        }
+    }
```

Comment:
I made two separate macros here mostly because I could not figure out how to match a pattern like:
```
Vec::<u8>
```
Any ideas?